src/execute.rs

```diff
@@ -426,10 +426,346 @@ impl Execute for System {
+        // LAX: LDA & LDX
+        let (value, cycles) = self.fetch_operand_value(opcode);
+        self.registers.a = value;
+        self.registers.x = value;
+        self.registers.sr.set_nz(value);
+
+        Ok(cycles)
+      }
+
+      0xC3 | 0xC7 | 0xCF | 0xD3 | 0xD7 | 0xDB | 0xDF => {
+        // DCP: DEC + SEC
+        self.registers.sr.set(flags::CARRY);
+
+        let (address, cycles) = self.fetch_operand_address(opcode);
+        let value = self.read(address);
+        let result = value.wrapping_sub(1);
+        self.registers.sr.set_nz(result);
+        self.write(address, result);
+
+        Ok(cycles + 2)
+      }
+
+      0xE3 | 0xE7 | 0xEF | 0xF3 | 0xF7 | 0xFB | 0xFF => {
+        // ISC: INC => SBC
+        let (address, cycles) = self.fetch_operand_address(opcode);
+        let value = self.read(address);
+        let result = value.wrapping_add(1);
+        self.registers.alu_subtract(value);
+        self.registers.sr.set_nz(result);
+        self.write(address, result);
+
+        Ok(cycles + 2)
+      }
+
+      0x0B | 0x2B => {
+        // ANC: AND byte with accumulator. If result is negative then carry is set.
+        let (value, cycles) = self.fetch_operand_value(opcode);
+        let new_val = self.registers.a & value;
+        self.registers.sr.write(flags::CARRY, new_val & 0x80 != 0);
+
+        Ok(cycles)
+      }
+
+      0x4B => {
+        // ALR: AND + LSR
+        let (value, cycles) = self.fetch_operand_value(opcode);
+        let new_val = (self.registers.a & value) >> 1;
+
+        self.registers.sr.write(flags::CARRY, new_val & 0x01 != 0);
+        self.registers.sr.set_nz(new_val);
+
+        Ok(cycles)
+      }
+
+      0x6B => {
+        // ARR: AND + ROR
+        let (value, cycles) = self.fetch_operand_value(opcode);
+        let mut new_val = self.registers.a & value;
+
+        let new_val = (new_val >> 1) | (self.registers.sr.read(flags::CARRY) as u8) << 7;
+
+        self.registers.sr.write(flags::CARRY, new_val & 0x40 != 0);
+        self
+          .registers
+          .sr
+          .write(flags::OVERFLOW, new_val & 0x20 != 0);
+        self.registers.sr.set_nz(new_val);
+
+        Ok(cycles)
+      }
+
+      0x8B => {
+        // XAA: AND X + AND immediate
+        // Oooo she's highly unstable xx "Do not use" or whatever
+
+        let (value, cycles) = self.fetch_operand_value(opcode);
+        let magic: u8;
+        #[cfg(not(target_arch = "wasm32"))]
+        {
+          magic = rand::random::<u8>();
+        }
+        #[cfg(target_arch = "wasm32")]
+        {
+          magic = 0xFF;
+        }
+        self.registers.a |= magic;
+        self.registers.a &= self.registers.x & value;
+        self.registers.sr.set_nz(self.registers.a);
+
+        Ok(cycles)
+      }
+
+      0xCB => {
+        // AXS: AND -> DEX -> STX
+        let (value, cycles) = self.fetch_operand_value(opcode);
+        self.registers.x &= self.registers.a;
+
+        self.registers.sr.set(flags::CARRY);
```

Comment:
```suggestion
        self.registers.alu_compare(self.registers.x, value);
```

The masswerk site lists this as "sets flags like CMP," which I think means we should set the N, V, and Z flags based on the comparison?

I think the `wrapping_sub` line is correct, given the "without borrow" part.